interactive/printer.go

```diff
@@ -2,42 +2,28 @@ package interactive
 
 import (
 	"fmt"
-	"io"
-	"strings"
+	"sort"
 
 	"github.com/viktomas/godu/core"
 )
 
-func PrintMarkedFiles(state *core.State, writer io.Writer) {
```

Comment:
I understand your point but the code is very complex for what it is trying to achieve. I'm proposing printing out all the files but sorting them by path string length, that means that if you pipe files to `mv` or `rm` you are always going to move/delete *child files first*.

I also think that when we allow user to mark the file (let's say subfolder of marked folder) we should put the file on the output because removal is only one of many possible use cases for marking files.